vernac/vernacextend.mli

```diff
@@ -156,14 +156,36 @@ type (_, _) ty_sig =
 
 type ty_ml = TyML : bool (* deprecated *) * ('r, 's) ty_sig * 'r * 's option -> ty_ml
 
-(** Wrapper to dynamically extend vernacular commands. *)
+(** Wrapper to dynamically extend vernacular commands.
+
+    This is used by coqpp VERNAC EXTEND.
+
+    Commands added by plugins at Declare ML Module / Require time should provide [plugin].
+
+    Commands added without providing [plugin] cannot be removed from
+    the grammar or modified. Not passing [plugin] is possible for
+    non-plugin coq-core commands and deprecated for all other callers.
+*)
 val vernac_extend :
   ?plugin:string ->
   command:string ->
   ?classifier:(string -> vernac_classification) ->
   ?entry:Vernacexpr.vernac_expr Pcoq.Entry.t ->
   ty_ml list -> unit
 
+(** For plugins which need more flexibility. The polymorphic arguments are as used in [TyML].
+    It is up to the caller to call [Egramml.extend_vernac_command_grammar] on the return value
+    to make the command grammar active.
```

Comment:
```suggestion
(** For grammar extensions which can be enabled dynamically (e.g. not at plugin
    loading time) and that can be disabled (e.g. by retracting over the command which
    enabled them).

    The polymorphic arguments are as in [TyML].
    The declared grammar extension is disabled, one needs to call
    [Egramml.extend_vernac_command_grammar] in order to enable it.
```

Reply:
Is "plugin loading time" well defined? I know "linking time" and "init time".

Reply:
linking time is better

Reply:
Trap question, plugins shouldn't extend the grammar at linking time

Reply:
My understanding is that they do, but it becomes active at import time.
Anyway, the comments should clarify this.

Reply:
>My understanding is that they do, but it becomes active at import time.

I guess that depends on what we mean by extend the grammar then.

BTW I think pushed an update while you were writing this comment.

Reply:
init_runtime looks reasonable